File: aib/server/src/routes/sessions.js
const express = require('express');
const router = express.Router();
const sessionService = require('../services/session');

// POST /api/sessions — Create a new session
router.post('/', async (req, res, next) => {
  try {
    const session = await sessionService.createSession();
    res.status(201).json(session);
  } catch (err) {
    next(err);
  }
});

// GET /api/sessions — List all sessions
router.get('/', async (req, res, next) => {
  try {
    const { status } = req.query;
    const sessions = await sessionService.listSessions(status);
    res.json(sessions);
  } catch (err) {
    next(err);
  }
});

// GET /api/sessions/:id — Get session details with messages
router.get('/:id', async (req, res, next) => {
  try {
    const session = await sessionService.getSessionWithMessages(req.params.id);
    if (!session) {
      return res.status(404).json({ error: 'Session not found' });
    }
    res.json(session);
  } catch (err) {
    next(err);
  }
});

module.exports = router;

File: aib/app/routes/sessions.py
"""
Session REST endpoints — mirrors server/src/routes/sessions.js
"""

from fastapi import APIRouter, HTTPException

from app.services import session as session_svc

router = APIRouter(prefix="/api/sessions", tags=["sessions"])


@router.post("", status_code=201)
async def create_session():
    """Create a new active session."""
    session = session_svc.create_session()
    return session


@router.get("")
async def list_sessions(status: str | None = None):
    """List all sessions, optionally filtered by status."""
    return session_svc.list_sessions(status)


@router.get("/{session_id}")
async def get_session(session_id: str):
    """Get session details with messages."""
    session = session_svc.get_session_with_messages(session_id)
    if not session:
        raise HTTPException(status_code=404, detail="Session not found")
    return session
